Data-Structures/src/main/java/sorting/MergeSort.java

```diff
@@ -0,0 +1,53 @@
+package sorting;
+
+public class MergeSort implements SortingAlgorithm {
+
+    // array to store values of original array to enable in place sorting of original array
+    private int[] aux;
+    // Sort array by recursively merging two sorted arrays together
+    @Override
+    public int[] sort(int[] arr) {

```

Comment:
You may want to consider making your sort a `static` method, that way you don't have to instantiate`MergeSort` objects to use it.